source/dubregistry/registry.d

```diff
@@ -388,7 +388,9 @@ class DubRegistry {
 
 	void downloadPackageZip(string packname, string vers, void delegate(scope InputStream) @safe del)
 	{
+		assert(vers.length > 0);
 		DbPackage pack = m_db.getPackage(packname);
+		assert(vers.length > 0);
```

Comment:
is it intended that this is 2x the same assert with no changes in value? There is nothing accessing vers there between the 2 calls.

If the stack got corrupted here it could possibly have any value other than 0 too so the assert would probably rarely trigger here. (the `length > 0` is the same as `length != 0` because it's unsigned so it's just checking for only 0) Also it's confusing for future code review why this assert exists 2x

Reply:
I agree on it being kind of confusing, maybe I'll just remove it. Of course *anything* can happen, like on Linux, where the symptoms were completely different. Aith the `alias this` for `Nullable.get` being removed soon, this is unlikely to slip back in, so the assertion doesn't have much chance of being an actual help in the future.